### PR TITLE
[placeholders]: Add regex capturing group as placeholders

### LINKED ISSUE #255: [placeholder] Add regex captoring group as placeholders
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
We currently use the capturing results for our zone lookup only.

**What you expected to happen**:
Use capturing results from regex (re) and simplified regex (from) to provide placeholders.
Placeholders proposed would be {1} basically the numbered matches and then also named matches {a} or {ab}.

**Anything else we need to know?**:
Problems with the named matches would be naming collisions, but that seems unlikely and we can add docs around reserved names for regex matches.

### PR DESCRIPTION
**Pull request by erbesharat**
**What this PR does / why we need it**:
Adds regex capturing group as placeholders

**Which issue this PR fixes**:
fixes #255

### CODE REVIEW

File: path.go [hunk 1]
```diff
@@ -42,9 +44,10 @@ func zoneFromPath(host string, path string, rec record) (string, int, []string,
 				unordered[group[1]] = pathSlice[i+1]
 			}
 			url := sortMap(unordered)
+			*r = *r.WithContext(context.WithValue(r.Context(), "regexMatches", unordered))
```
[the remark this answers is not in the document]
Comment (erbesharat) [reply path.go]: Well, I thought since the subdomain order is different than the path, we should use the unordered slice that is same as the path's order.

**Comment by erbesharat** [comment]
Fixed the conflict and updated the comment.

P.S for GitHub: I didn't dismiss anything :angry: :sweat_smile:

File: path.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply path.go]: That makes sense. Probably what was problematic was the naming. As it's not unordered, just not reordered yet.

**Pull request merged**